### PR TITLE
New relay mode (with postcard)

### LINKED ISSUE #66: error in grpc streaming ingest_measurements
**Describe the bug**
On a ovh k8s cluster, I deployed an ALUMet client on each cluster's node (2) and 1 ALUMet server. Sometimes, few second after the deployment (using an helm chart), all ALUMet's client crashed with the following error:
ERROR alumet::pipeline::elements::output] Error when asynchronously writing to relay-client/output/grpc-measurements (will stop running): error in grpc streaming ingest_measurements: status status: Cancelled, message: "h2 protocol error: http2 error", details: [], metadata: MetadataMap { headers: {} }

**Environment information**
OS: k8s on ovh
Alumet version²: 0.6.1 + energy-estimation-tdp plugin
Enabled plugins: 
- EnergyEstimationTdpPlugin v0.1.0
        - k8s v0.1.0
        - procfs v0.1.0
        - relay-client v0.4.0
        - socket-control v0.2.0

²: please include the commit hash if it's not a release
2611b112e82d43a7eb267c391bca1ce2d1b6f66f

**To Reproduce**
1. deploy the ALUMet helm chart but the error is not reproductable every time ! I reproduced at least 3 times on ovh k8s cluster. 

I did several deploiement tests this afternoon on ovh ck8s luster and the results are:

1. 16h03: reproduce the error,
2. 16h18: no error
3. 16h27: reproduce the error,
4. 16h31: no error
5. 14h41: no error
6. 16h57: no error
7. 17h05: no error

**Expected behavior**
the ALUMet pipepline crashed, we get a grpc error

**Logs / Output**
Attached logs file of server and client part.
[alumet-client-ovh-demo-ns-demo-test-2024-11-13.txt](https://github.com/user-attachments/files/17871358/alumet-client-ovh-demo-ns-demo-test-2024-11-13.txt)
[alumet-server-ovh-demo-ns-demo-test-2024-11-13.txt](https://github.com/user-attachments/files/17871361/alumet-server-ovh-demo-ns-demo-test-2024-11-13.txt)
Below a screenshot influxdb dashboard but not at the same times as the above log files
influxdb-crash-AL-client

### PR DESCRIPTION
**Pull request by TheElectronWill**
This should fix #66, and vastly improve the performance of the relay mode (~3% cpu usage at 1kHz vs 9% with the old relay mode).

### CODE REVIEW

File: app-agent/src/bin/relay_client.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ fn main() {
         plugin_perf::PerfPlugin,
         plugin_socket_control::SocketControlPlugin,
         plugin_relay::client::RelayClientPlugin,
-        plugin_cgroupv2::K8sPlugin,
+        // plugin_cgroupv2::K8sPlugin,
```
Comment (titouanj) [inline app-agent/src/bin/relay_client.rs:20]: ?

File: app-agent/tests/relay_together.rs [hunk 2]
```diff
@@ -15,33 +15,32 @@ mod common;
 ///
 /// Note: we use a limited set of plugins so that it works in the CI environment.
 #[test]
-fn client_to_server_to_csv_ipv4() {
-    // works in CI
-    client_to_server_to_csv_on_address(Some(("localhost", "50051"))).unwrap()
-}
+fn client_to_server_to_csv() {
+    // These tests are in the same test function because they must NOT run concurrently (same port).
```
Comment (titouanj) [inline app-agent/tests/relay_together.rs:19]: You should do the same thing for local.rs and relay_server.rs to fix the problems about Cargo.lock

**Comment by TheElectronWill** [comment]
This PR will also fix https://github.com/oar-team/nur-kapack/blob/4c398cb11980ae8429ed51f14f079eff80205c0d/pkgs/alumet/default.nix#L40

File: .circleci/config.yml [hunk 3]
```diff
@@ -41,13 +41,20 @@ jobs:
...
+      # build libraries and agents now to avoid a timeout in the tests
       - run:
           name: Build all libraries
           command: "cargo build"
+      - run:
+          name: Build the agents
+          command: |
+            cargo build --bin alumet-local-agent --features local_x86
+            cargo build --bin alumet-relay-client --features relay_client
+            cargo build --bin alumet-relay-server --features relay_server
```
[the remark this answers is not in the document]
Comment (TheElectronWill) [reply .circleci/config.yml]: Today I learned something :) thanks

File: app-agent/tests/common/run.rs
Comment (titouanj) [inline app-agent/tests/common/run.rs:1]: Wonderful!

**Review by titouanj — APPROVED** [review]
Must be rebased on main

**Pull request merged**